7z, 7za, 7zr: updates (#5432)
tldr-pages/tldr: pages/common/7z.md

# 7z

> File archiver with a high compression ratio.
> More information: <https://www.7-zip.org>.

> - Archive a file or directory:

Reviewer: ```suggestion
- [a]rchive a file or directory:
```

`7z a {{path/to/archive.7z}} {{path/to/file_or_directory}}`

> - Encrypt an existing archive (including file names):

Reviewer 2: Perhaps this will read better?
```suggestion
- Encrypt an existing archive (including filenames):
```

`7z a {{path/to/encrypted.7z}} -p{{password}} -mhe=on {{path/to/archive.7z}}`

> - Extract an archive preserving the original directory structure:

Reviewer: ```suggestion
- E[x]tract an archive preserving the original directory structure:
```

`7z x {{path/to/archive.7z}}`

> - Extract an archive to a specific directory:

Reviewer: ```suggestion
- E[x]tract an archive to a specific directory:
```

`7z x {{path/to/archive.7z}} -o{{path/to/output}}`

> - Extract an archive to stdout:

Reviewer: ```suggestion
- E[x]tract an archive to stdout:
```

`7z x {{path/to/archive.7z}} -so`

> - Archive using a specific archive type:

Reviewer: ```suggestion
- [a]rchive using a specific archive type:
```

`7z a -t{{zip|gzip|bzip2|tar}} {{path/to/archive.7z}} {{path/to/file_or_directory}}`

> - List the contents of an archive:

Reviewer: ```suggestion
- [l]ist the contents of an archive:
```

> `7z a -t{{7z|zip|gzip|bzip2|lzip}} {{path/to/archive.7z}} {{path/to/file_or_directory}}`

Reviewer 2: What's `lzip`, and why is it replacing `tar`? `7z i | grep lzip` comes up with nothing.

Author: Hmmm. I got it from the man page:

>Type of archive (7z, zip, gzip, bzip2, lzip or tar. 7z format is default)

<details>
<summary>Output of `7z i` on macOS 11.2.3</summary>

```
7-Zip [64] 17.04 : Copyright (c) 1999-2021 Igor Pavlov : 2017-08-28
p7zip Version 17.04 (locale=utf8,Utf16=on,HugeFiles=on,64 bits,16 CPUs x64)

Libs:
 0  /usr/local/Cellar/p7zip/17.04/lib/p7zip/7z.dll

Formats:
 0 C   F         7z       7z            7 z BC AF ' 1C
 0               APM      apm           E R
 0               Ar       ar a deb lib  ! < a r c h > 0A
 0               Arj      arj           ` EA
 0 CK            bzip2    bz2 bzip2 tbz2 (.tar) tbz (.tar) B Z h
 0     F         Cab      cab           M S C F 00 00 00 00
 0               Chm      chm chi chq chw I T S F 03 00 00 00 ` 00 00 00
 0     F         Hxs      hxs hxi hxr hxq hxw lit I T O L I T L S 01 00 00 00 ( 00 00 00
 0               Compound msi msp doc xls ppt D0 CF 11 E0 A1 B1 1A E1
 0      M        Cpio     cpio          0 7 0 7 0  ||  C7 q  ||  q C7
 0               CramFS   cramfs        offset=16 C o m p r e s s e d 20 R O M F S
 0       G  B    Dmg      dmg           k o l y 00 00 00 04 00 00 02 00
 0           E   ELF      elf            E L F
 0               Ext      ext ext2 ext3 ext4 img offset=1080 S EF
 0               FAT      fat img       offset=510 U AA
 0               FLV      flv           F L V 01
 0 CK            gzip     gz gzip tgz (.tar) tpz (.tar) apk (.tar) 1F 8B 08
 0               GPT      gpt mbr       offset=512 E F I 20 P A R T 00 00 01 00
 0      M        HFS      hfs hfsx      offset=1024 H + 00 04  ||  H X 00 05
 0        O      IHex     ihex          
 0               Iso      iso img       offset=32769 C D 0 0 1
 0               Lzh      lzh lha       offset=2 - l h
 0  K     O      lzma     lzma          
 0  K            lzma86   lzma86        
 0      M    E   MachO    macho         CE FA ED FE  ||  CF FA ED FE  ||  FE ED FA CE  ||  FE ED FA CF
 0         P     MBR      mbr           
 0               MsLZ     mslz          S Z D D 88 F0 ' 3 A
 0      M        Mub      mub           CA FE BA BE 00 00 00  ||  B9 FA F1 0E
 0     F G       Nsis     nsis          offset=4 EF BE AD DE N u l l s o f t I n s t
 0               NTFS     ntfs img      offset=3 N T F S 20 20 20 20 00
 0           E   PE       exe dll sys   M Z
 0           E   TE       te            V Z
 0               Ppmd     pmd           8F AF AC 84
 0               QCOW     qcow qcow2 qcow2c Q F I FB 00 00 00
 0     F         Rar      rar r00       R a r ! 1A 07 00
 0     F         Rar5     rar r00       R a r ! 1A 07 01 00
 0               Rpm      rpm           ED AB EE DB
 0               Split    001           
 0      M        SquashFS squashfs      h s q s  ||  s q s h  ||  s h s q  ||  q s h s
 0 C    M        SWFc     swf (~.swf)   C W S  ||  Z W S
 0  K            SWF      swf           F W S
 0 C      O   LH tar      tar ova       offset=257 u s t a r
 0        O      Udf      udf iso img   offset=32768 01 C D 0 0 1
 0     FM        UEFIc    scap          BD 86 f ; v 0D 0 @ B7 0E B5 Q 9E / C5 A0  ||  8B A6 < J # w FB H 80 = W 8C C1 FE C4 M  ||  B9 82 91 S B5 AB 91 C B6 9A E3 A9 C F7 / CC
 0     FM        UEFIf    uefif         offset=16 D9 T 93 z h 04 J D 81 CE 0B F6 17 D8 90 DF  ||  x E5 8C 8C = 8A 1C O 99 5 89 a 85 C3 - D3
 0               VDI      vdi           offset=64  10 DA BE
 0       G       VHD      vhd           c o n e c t i x 00 00
 0               VMDK     vmdk          K D M V
 0 C SN       LH wim      wim swm esd ppkg M S W I M 00 00 00
 0               Xar      xar pkg xip   x a r ! 00 1C
 0 CK            xz       xz txz (.tar) FD 7 z X Z 00
 0               Z        z taz (.tar)  1F 9D
 0 C   FMG       zip      zip z01 zipx jar xpi odt ods docx xlsx epub ipa apk appx P K 03 04  ||  P K 05 06  ||  P K 06 06  ||  P K 07 08 P K  ||  P K 0 0 P K
 0 CK            lizard   liz tliz (.tar) 0 x 1 8 4 D 2 2 0 6 00
 0  K            lzip     lz tlz (.tar) L Z I P 01
 0 CK            lz4      lz4 tlz4 (.tar) 0 x 1 8 4 D 2 2 0 4 00
 0 CK            lz5      lz5 tlz5 (.tar) 0 x 1 8 4 D 2 2 0 5 00
 0 CK            zstd     zst tzstd (.tar) 0 x F D 2 F B 5 2 2 . . 2 8 00

Codecs:
 0  ED    40202 BZip2
 0 4ED  303011B BCJ2
 0  ED  3030103 BCJ
 0  ED  3030205 PPC
 0  ED  3030401 IA64
 0  ED  3030501 ARM
 0  ED  3030701 ARMT
 0  ED  3030805 SPARC
 0  ED    20302 Swap2
 0  ED    20304 Swap4
 0  ED        0 Copy
 0  ED    40109 Deflate64
 0  ED    40108 Deflate
 0  ED        3 Delta
 0  ED       21 LZMA2
 0  ED    30101 LZMA
 0  ED    30401 PPMD
 0  ED       21 FLZMA2
 0  ED  4F71106 LIZARD
 0  ED  4F71104 LZ4
 0  ED  4F71105 LZ5
 0  ED  4F71101 ZSTD
 0  ED  4F71102 BROTLI
 0  ED  6F10701 7zAES
 0  ED  6F00181 AES256CBC

Hashers:
 0   32      202 BLAKE2sp
 0    4        1 CRC32
 0   20      201 SHA1
 0   32        A SHA256
 0    8        4 CRC64
 0   16      205 MD2
 0   16      206 MD4
 0   16      207 MD5
 0   48      208 SHA384
 0   64      209 SHA512
 0    4      203 XXH32
 0    8      204 XXH64
```

 <details>

Reviewer 3: on linux:
```bash
$ 7za i
...
   K            lzip     lz tlz (.tar) L Z I P 01
```

Author: Well, I'm on a Mac so it might be different.  How about just keeping it as is?  For a user, knowing that it's a possibility when it isn't seems better than not know though it is.

Reviewer: @bl-ue maybe that's because you're using version `16.02` of `7z`. I have version `17.04` and it's showing up.

Reviewer 2: Ah, I see

That must be the issue @marchersimon. Let's keep this as-is then.
